### DISCUSSION TITLE
Problem with playlist behavior and shuffle.

### DISCUSSION THREAD

**Discussion by ClockWorkAsylum** [smplayer-dev/smplayer#826] [General]
Ive noticed an odd issue with the playlist.

when the playlist is sorted by the position column, the shuffle no longer works.

any help is appreciated

**Comment by smplayer-dev** [comment]
I think this is normal. The shuffle function creates a hidden column with random numbers and sorts the list by that column. If you click in another column the sort is changed and shuffle doesn't work.

**Reply by ClockWorkAsylum** [reply]
i know the 19.10 version has this feature. the oldest i can compile from here is 20..  do you know where i can find the 19.10 source?

**Reply by smplayer-dev** [reply]
https://sourceforge.net/projects/smplayer/files/SMPlayer/

**Reply by ClockWorkAsylum** [reply]
ive tried to build from source but get a number of errors, i think related to the qt changes in naming i.e.. qt5-default changed to qtbase5-dev.. 

i very much want to run the 19.10 version and i know little of coding..  
which smplayer forum would you suggest i ask for assistance??

**Reply by smplayer-dev** [reply]
For Ubuntu, these commands should compile and create a deb package:

```
sudo apt-get install zlib1g-dev fakeroot build-essential devscripts debhelper g++ qtbase5-dev qt5-qmake qtscript5-dev qttools5-dev-tools qtbase5-private-dev --no-install-recommends

./create_deb.sh
```

**Reply by ClockWorkAsylum** [reply]
thank you so much for your time, it was much appreciated. Unfortunately the instructions you provided produced the same errors as following the included install.txt file.

**Reply by smplayer-dev** [reply]
What errors?

**Reply by ClockWorkAsylum** [reply]
[error.txt](https://github.com/smplayer-dev/smplayer/files/13312714/error.txt)

im not sure where the log files are for the build so here are the last few lines from the console. its cut/paste, sorry.

**Reply by smplayer-dev** [reply]
Try copying this file to the webserver folder:
https://raw.githubusercontent.com/smplayer-dev/smplayer/master/webserver/mongoose.h

**Reply by ClockWorkAsylum** [reply]
The build failed at the same place. 

I've decided to go back to ubuntu 20. 

Thanks again for your time.

**Reply by ClockWorkAsylum** [reply]
UPDATE
your change to the mongoose.h file did work. the problem was in the package creation. i reverted to make all and used checkinstall to install and create the package. seems to work great so far.. 

again, thanks for the help!
